### ISSUE TITLE
[llCompiler-58] System abi load statement

### ISSUE THREAD

**Issue by LarsBehl** [LarsBehl/llCompiler#124]
As a user I want to be able to use functions and types exposed by another library. This also includes libraries written for `C`. For libraries written in `LL` a header like file will pre present (see #125). In order for `C` libraries to be usable without compile time errors, there has to be a way to declare functions as imported form a native library.

Blocked by #125 

## Acceptance Criteria
* [ ] An additional keyword `native` for function prototypes is implemented
* [ ] Using native functions does not generate compiletime errors
* [ ] Using native functions does not generate runtime errors, as long as the lib is linked to the program

## Background
In order to be able to implement a base class library, loading and working with default libraries like the `glibc` is needed.

The first idea was to locate the needed headers (like `sys/syscall.h`) and parse them. To do this, another parser has to be implemented in order to collect the function definitions in the form needed by `LL`. This would be a time consuming task. To speed up the development a special system-load-statement was brought up. Using this statement a user could declare a function as exported by the system or another library. These function prototyes would be collected and stored. This would prevent any compile time errors occuring when trying to call a function that is not defined.

This is definitely not the most elegant way to implement the usage of abis, but it enables the developers to speed up the development of the base class library.

Another possibility to not rely on the `glibc` at all. One could implement systemcalls by being compliant to the convention used by the operating system and architecture. Modern `x86` CPUs offer a instruction called `syscall` which the OS implements a handler for and *tries* to handle the requested system call. Using this method an inline assembly statement would be a great choice, as the wrapper methods for the system calls could be directly implemented in `LL`.
However, the literature read so far does not encourage implementing systemcalls manually. It is recommended to use libraries like `glibc`. Using libraries also offers the possibility to run the compiled code on operating systems using different conventions for system calls

**Comment by LarsBehl** [comment]
After a conversation with my professor today, we decided that using the `glibc` as a basis is probably the best way. To be able to use functions supplied by the `glibc` and other `C` libraries, the definition of prototypes (see #125) has to be extended by an additional keyword to specify that a function is imported from a native context. Therefor the keyword `native` is probably a good choice

**Comment by LarsBehl** [comment]
After beeing close to finishing the header files implementation a question arised: With the current implementation of prototypes the keyword `native` won't be needed. It would be sufficient to supply header files for the libraries that are getting used like `glibc`. The compiler is not doing any assumptions about the defined functions or types, it simply registers them as available prototypes. When a prototype function is called, the needed code for the function call is generated. If no implementation of the function is present, the compiler and linker for the assembler code is the first instance that could throw an error or warning when, as an unknown method is referenced. I will move this issue to `On hold` until the implementation of #125 is finished and a first test with referencing methods present in a binary file is conducted
